components/raftstore-v2/src/raft/peer.rs

```diff
@@ -1,47 +1,47 @@
 // Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
 
-use engine_traits::RaftEngine;
-use kvproto::metapb;
-use raft::RawNode;
-use raftstore::store::Config;
+use std::sync::Arc;
+
+use engine_traits::{KvEngine, RaftEngine, TabletFactory};
+use kvproto::{metapb, raft_serverpb::RegionLocalState};
+use raft::{RawNode, INVALID_ID};
+use raftstore::store::{util::find_peer, Config};
 use slog::{o, Logger};
 use tikv_util::{box_err, config::ReadableSize};
 
 use super::storage::Storage;
 use crate::Result;
 
 /// A peer that delegates commands between state machine and raft.
-pub struct Peer<ER: RaftEngine> {
-    region_id: u64,
-    peer: metapb::Peer,
+pub struct Peer<EK: KvEngine, ER: RaftEngine> {
```

Comment:
Let `PeerFsm` stores the states and only provides some non-trival getter and setter. And moves all the message handling logic to `PeerFsmDelegate` to make strong cohesion, and category the logics in separated files.

Reply:
Ideally, `Peer` is a wrapper. But for the current impl, we pass `ctx` all way down to the functions of `Peer` when handling messages in `PeerFsm`, it mixes the logic with `Peer` because states are mostly located in `Peer`.

Reply:
`ctx` is different from messages. It's just a way to make the function parameter less.